### PR TITLE
Enable using awsx provider, other providers with shallow plain resource properties

### LINKED ISSUE #177: Replace code generation tests with `pulumi convert`

### LINKED ISSUE #193: Resource in awsx provider expects plain value, errors due to receiving Output
Using the same YAML program as in #184, an error occurs on preview due to the `taskDefinitionArgs` property being an Output, when it should be plain with sub-sub-properties allowed to be Outputs. 

See this related bug and change in the Go SDK codegen to allow shallow plain values: 
- https://github.com/pulumi/pulumi/issues/9507

Program and output:

```
name: awsx
runtime: yaml
resources:
  cluster:
    type: aws:ecs:Cluster
  lb:
    type: awsx:lb:ApplicationLoadBalancer
  nginx:
    type: awsx:ecs:FargateService
    properties:
      cluster: ${cluster.arn}
      taskDefinitionArgs:
        container:
          image: nginx:latest
          cpu: 512
          memory: 128
          portMappings:
            - containerPort: 80
              targetGroup: ${lb.defaultTargetGroup}
outputs: omitted here, separate bug
  url: ${lb.loadBalancer.dnsName}
```

```
pulumi up
Previewing update (dev)

View Live: https://app.pulumi.com/friel/awsx/dev/previews/84427b80-1851-4647-9030-bd8d2b674602

     Type                                  Name      Plan       Info
 +   pulumi:pulumi:Stack                   awsx-dev  create     1 error; 11 messages
 +   ├─ aws:ecs:Cluster                    cluster   create     
 +   ├─ awsx:lb:ApplicationLoadBalancer    lb        create     
 +   │  ├─ aws:ec2:SecurityGroup           lb        create     
 +   │  ├─ aws:lb:TargetGroup              lb        create     
 +   │  ├─ aws:lb:LoadBalancer             lb        create     
 +   │  └─ aws:lb:Listener                 lb-0      create     
 +   └─ awsx:ecs:FargateService            nginx     create     
 +      └─ awsx:ecs:FargateTaskDefinition  nginx     create     
 
Diagnostics:
  pulumi:pulumi:Stack (awsx-dev):
    Error: Exactly one of [container] or [containers] must be provided: Error: Exactly one of [container] or [containers] must be provided
        at normalizeFargateTaskDefinitionContainers (/home/friel/.pulumi/plugins/ecs/fargateTaskDefinition.ts:120:15)
        at new FargateTaskDefinition (/home/friel/.pulumi/plugins/ecs/fargateTaskDefinition.ts:60:28)
        at new FargateService (/home/friel/.pulumi/plugins/ecs/fargateService.ts:55:30)
        at awsx:ecs:FargateService (/home/friel/.pulumi/plugins/resources.ts:26:45)
        at construct (/home/friel/.pulumi/plugins/resources.ts:56:12)
        at Provider.construct (/home/friel/.pulumi/plugins/index.ts:43:35)
        at Server.<anonymous> (/home/friel/.pulumi/plugins/resource-awsx-v1.0.0/node_modules/@pulumi/provider/server.ts:322:48)
        at Generator.next (<anonymous>)
        at fulfilled (/home/friel/.pulumi/plugins/resource-awsx-v1.0.0/node_modules/@pulumi/pulumi/provider/server.js:18:58)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

### PR DESCRIPTION
**Pull request by AaronFriel**
Modify `evaluateObject` to keep maps plain as long as their keys are plain. This is equivalent to allowing "shallow plain" properties in YAML, and modify `evaluateAccessProperties` so that resource references are hydrated via built-in invoke.

Revert a commit that changed dependency tracking to add items to the `DependsOn` resource option and instead relies on an upstream change (pulumi/pulumi#9566) to ensure that `RawOutputs` has the right dependency of the resource. This works around a bug in the Node SDK documented here here:
- https://github.com/pulumi/pulumi/issues/9567

With these changes, we retain precise tracking of the dependencies *and* support shallow plain resource properties, while working around the bug in the Node SDK.

This unblocks being able to run the awsx example.

Addendum:

Integration tests now pass and codegen for all four languages. Type checking/building fails for NodeJS (PR merged, waiting on release), and Go (issue here: https://github.com/pulumi/pulumi/issues/9664).

Tests were updated to use `GenerateProject`, just as Pulumi convert does, and to not mutate/modify the files when type checking/validating them.

Fixes #193.
Fixes #177.

### CODE REVIEW

**Review by iwahbe — CHANGES_REQUESTED** [review]
I think the changes to dependency tracking look good. I don't like that we are adding a failing test. I would like us to ensure that master is always green.

File: pkg/tests/example_transpile_test.go [hunk 1]
```diff
@@ -121,15 +131,18 @@ func TestGenerateExamples(t *testing.T) {
 	// memory consumption of using independent schema.Package instances, we
 	// decided to maintain a single copy of each schema until non-determinism is
 	// observed.
-	var generateLock sync.Mutex
```
Comment (iwahbe) [inline pkg/tests/example_transpile_test.go:124]: If your going to dispose of the lock:
1. I want to know what has changed that this won't cause occasional race conditions (and thus flakey tests).
2. We should remove the comment explaining the lock if we remove the lock.

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -19,3 +19,6 @@
 - Allow Fn::ToBase64 to take expressions as inputs, was previously constrained to only allow a
   string constant.
   [#221](https://github.com/pulumi/pulumi-yaml/pull/221)
+
+- [expr] Fix handling of "plain" input maps when sending properties to component providers such as AWSX.
+  https://github.com/pulumi/pulumi-yaml/pull/195
```
Comment (iwahbe) [inline CHANGELOG_PENDING.md:24]: We seem to have a format for the changelog. I would prefer we remain consistent:
```suggestion
  [#195](https://github.com/pulumi/pulumi-yaml/pull/195)
```

**Review by iwahbe — APPROVED** [review]
Some nits, but in general LGTM

File: pkg/tests/example_transpile_test.go (hunk 1 above)
Comment (AaronFriel) [reply pkg/tests/example_transpile_test.go:124]: The primary issue with the lock was `yarn install`, and I kept it for wrapping the `TypeCheck` function exclusively.

Agree on removing the comment :)

Comment (AaronFriel) [reply pkg/tests/example_transpile_test.go:124]: I also think my related changes upstream & to use GenerateProject solved the concurrency issues, could not reproduce locally with high `-parallelism`.

Comment (iwahbe) [reply pkg/tests/example_transpile_test.go:124]: I remember that it had something to do with binding the same schema in parallel, but your inability to replicate indicates that I misremembered.

**Pull request merged**